ATB.Tests/BookingServiceTests.cs

```diff
@@ -0,0 +1,297 @@
+using ATB.DataAccess;
+using ATB.Entities;
+using ATB.Services;
+using AutoFixture;
+using AutoFixture.AutoMoq;
+using Moq;
+
+namespace ATB.Tests
+{
+    public class BookingServiceTests
+    {
+        private readonly IFixture fixture;
+        private readonly Mock<IBookingRepository> mockBookingRepository;
+        private readonly BookingService sut;
+        public BookingServiceTests()
+        {
+            fixture = new Fixture().Customize(new AutoMoqCustomization());
+            mockBookingRepository = fixture.Freeze<Mock<IBookingRepository>>();
+            sut = new BookingService(mockBookingRepository.Object);
+        }
+
+        [Fact]
+        public void GetAllBookings_ReturnsAllBookings()
+        {
+            var bookings = fixture.CreateMany<Booking>();
+            mockBookingRepository.Setup(x => x.GetAllBookings()).Returns(bookings);
+
+            var result = sut.GetAllBookings();
+
+            Assert.Equal(bookings, result);
+        }
+
+        [Fact]
+        public void AddBooking_PassengerHasNotBookedThisFlightBefore_ReturnsSuccess()
+        {
+            var passenger = fixture.Create<Passenger>();
+            var flight = fixture.Create<Flight>();
+            var flightClass = fixture.Create<FlightClass>();
+            var booking = new Booking(flight, passenger, flightClass);
+            var bookings = new List<Booking>();
+
+            mockBookingRepository.Setup(x => x.AddBooking(booking));
+            mockBookingRepository.Setup(x => x.GetAllBookings()).Returns(bookings);
+
+            var result = sut.AddBooking(passenger, flight, flightClass);
+
+            Assert.Equal(BookingOperationStatus.Success, result);
+            mockBookingRepository.Verify(x => x.AddBooking(booking), Times.Once());
+
+        }
+        [Fact]
+        public void AddBooking_PassengerHasBookedThisFlightBefore_ReturnsFailed()
+        {
+            var passenger = fixture.Create<Passenger>();
+            var flight = fixture.Create<Flight>();
+            var flightClass = fixture.Create<FlightClass>();
+            var booking = new Booking(flight, passenger, flightClass);
+            var bookings = new List<Booking>
+            {
+                booking
+            };
```

Comment:
can we use auto fixture for those initialization?

Reply:
var booking = fixture.Build<Booking>()
                .With(b => b.Flight, flight)
                .With(b => b.Passenger, passenger)
                .With(b => b.FClass, flightClass)
                .Create();